tools/stylelint/no-ampersand-beyond-selector-start.ts

```diff
@@ -37,43 +36,38 @@ const plugin = createPlugin(ruleName, (isEnabled: boolean, _options?) => {
     }
 
     root.walkRules(rule => {
-      if (
-        rule.parent.type === 'rule' &&
-        isStandardSyntaxRule(rule) &&
-        isStandardSyntaxSelector(rule.selector) &&
-        // Using the ampersand at the beginning is fine, anything else can cause issues in themes.
-        rule.selector.indexOf('&') > 0) {
-
-        const mixinName = getClosestMixinName(rule);
-
-        // Skip rules inside private mixins.
-        if (!mixinName || !mixinName.startsWith('_')) {
-          utils.report({
-            result,
-            ruleName,
-            message: messages.expected(),
-            node: rule
-          });
-        }
+      if (rule.parent.type === 'rule' &&
+          isStandardSyntaxRule(rule) &&
+          isStandardSyntaxSelector(rule.selector) &&
+          hasInvalidAmpersandUsage(rule.selector)) {
+        utils.report({
+          result,
+          ruleName,
+          message: messages.expected(),
+          node: rule
+        });
       }
     });
   };
+});
+
+function hasInvalidAmpersandUsage(selector: string): boolean {
+  const parts = selector.split(',').map(part => part.trim());
 
-  /** Walks up the AST and finds the name of the closest mixin. */
-  function getClosestMixinName(node: Node): string | undefined {
-    let parent = node.parent;
+  for (const part of parts) {
+    let index = part.indexOf('&');
```

Comment:
Would it be sufficient to just check `part.indexOf('&', 1)` instead of having the while loop?